java/memory/src/main/java/org/apache/arrow/memory/LowCostIdentityHasMap.java

```diff
@@ -18,44 +18,38 @@
 package org.apache.arrow.memory;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
 
 /**
  * Highly specialized IdentityHashMap that implements only partial
  * Map APIs
  * It incurs low initial cost (just two elements by default)
  * It assumes Value includes the Key - Implements @ValueWithKeyIncluded iface
  * that provides "getKey" method
+ * @param <K>
  * @param <V>
  */
-public class LowCostIdentityHasMap<V extends ValueWithKeyIncluded> {
+public class LowCostIdentityHasMap<K, V extends ValueWithKeyIncluded<K>> {
```

Comment:
No need to declare K since V already declares it.

Reply:
done